### ISSUE TITLE
fs.writeFile fails when file >= 1024 bytes

### ISSUE THREAD

**Issue by yetzt** [tessel/sdcard#25]
when i try to write a file with 1024 bytes or more, i get

`Error: No more tries left waiting for idle.`

i cranked up the debug level and tested with 1023 bytes:

```
writing ikb8u9wi.bin with 1023 bytes
** 844 ----- SPI QUEUE REQUESTED ----- #7
** 32 ----- SPI QUEUE ACQUIRED ----- #7
** 19 sendCommand READ_SINGLE_BLOCK 11948
** 11 _sendCommand 17 0x2eac
** 12 * sending data: <Buffer 51 00 00 2e ac 00>
** 44 TRANSFER RESULT <Buffer ff ff ff ff ff ff>
** 44 while waiting for response got <Buffer ff>
** 40 while waiting for response got <Buffer 00>
** 41 While waiting for data, got 0xff on try 100
** 31 While waiting for data, got 0xfe on try 99
** 59 ----- RELEASING SPI QUEUE ----- #7
** 453 ----- SPI QUEUE REQUESTED ----- #8
** 32 ----- SPI QUEUE ACQUIRED ----- #8
** 21 sendCommand READ_SINGLE_BLOCK 11949
** 11 _sendCommand 17 0x2ead
** 13 * sending data: <Buffer 51 00 00 2e ad 00>
** 53 TRANSFER RESULT <Buffer ff ff ff ff ff ff>
** 64 while waiting for response got <Buffer ff>
** 64 while waiting for response got <Buffer 00>
** 82 While waiting for data, got 0xff on try 100
** 67 While waiting for data, got 0xfe on try 99
** 121 ----- RELEASING SPI QUEUE ----- #8
** 221 ----- SPI QUEUE REQUESTED ----- #9
** 87 ----- SPI QUEUE ACQUIRED ----- #9
** 41 sendCommand READ_SINGLE_BLOCK 11950
** 18 _sendCommand 17 0x2eae
** 19 * sending data: <Buffer 51 00 00 2e ae 7f>
** 48 TRANSFER RESULT <Buffer ff ff ff ff ff ff>
** 42 while waiting for response got <Buffer ff>
** 41 while waiting for response got <Buffer 00>
** 41 While waiting for data, got 0xff on try 100
** 35 While waiting for data, got 0xfe on try 99
** 57 ----- RELEASING SPI QUEUE ----- #9
** 111 ----- SPI QUEUE REQUESTED ----- #10
** 29 ----- SPI QUEUE ACQUIRED ----- #10
** 19 sendCommand READ_SINGLE_BLOCK 11951
** 11 _sendCommand 17 0x2eaf
** 11 * sending data: <Buffer 51 00 00 2e af 8f>
** 43 TRANSFER RESULT <Buffer ff ff ff ff ff ff>
** 40 while waiting for response got <Buffer ff>
** 38 while waiting for response got <Buffer 00>
** 41 While waiting for data, got 0xff on try 100
** 30 While waiting for data, got 0xfe on try 99
** 57 ----- RELEASING SPI QUEUE ----- #10
** 112 ----- SPI QUEUE REQUESTED ----- #11
** 30 ----- SPI QUEUE ACQUIRED ----- #11
** 18 sendCommand READ_SINGLE_BLOCK 11952
** 11 _sendCommand 17 0x2eb0
** 11 * sending data: <Buffer 51 00 00 2e b0 00>
** 42 TRANSFER RESULT <Buffer ff ff ff ff ff ff>
** 40 while waiting for response got <Buffer ff>
** 39 while waiting for response got <Buffer 00>
** 40 While waiting for data, got 0xff on try 100
** 30 While waiting for data, got 0xfe on try 99
** 57 ----- RELEASING SPI QUEUE ----- #11
** 142 ----- SPI QUEUE REQUESTED ----- #12
** 29 ----- SPI QUEUE ACQUIRED ----- #12
** 19 sendCommand READ_SINGLE_BLOCK 11953
** 10 _sendCommand 17 0x2eb1
** 12 * sending data: <Buffer 51 00 00 2e b1 ff>
** 42 TRANSFER RESULT <Buffer ff ff ff ff ff ff>
** 40 while waiting for response got <Buffer ff>
** 38 while waiting for response got <Buffer 00>
** 40 While waiting for data, got 0xff on try 100
** 30 While waiting for data, got 0xfe on try 99
** 57 ----- RELEASING SPI QUEUE ----- #12
** 631 ----- SPI QUEUE REQUESTED ----- #13
** 29 ----- SPI QUEUE ACQUIRED ----- #13
** 19 sendCommand READ_SINGLE_BLOCK 11954
** 10 _sendCommand 17 0x2eb2
** 11 * sending data: <Buffer 51 00 00 2e b2 ff>
** 43 TRANSFER RESULT <Buffer ff ff ff ff ff ff>
** 39 while waiting for response got <Buffer ff>
** 39 while waiting for response got <Buffer 00>
** 40 While waiting for data, got 0xff on try 100
** 30 While waiting for data, got 0xfe on try 99
** 56 ----- RELEASING SPI QUEUE ----- #13
** 631 ----- SPI QUEUE REQUESTED ----- #14
** 30 ----- SPI QUEUE ACQUIRED ----- #14
** 18 sendCommand READ_SINGLE_BLOCK 11955
** 10 _sendCommand 17 0x2eb3
** 12 * sending data: <Buffer 51 00 00 2e b3 10>
** 42 TRANSFER RESULT <Buffer ff ff ff ff ff ff>
** 40 while waiting for response got <Buffer ff>
** 38 while waiting for response got <Buffer 00>
** 40 While waiting for data, got 0xff on try 100
** 30 While waiting for data, got 0xfe on try 99
** 56 ----- RELEASING SPI QUEUE ----- #14
** 633 ----- SPI QUEUE REQUESTED ----- #15
** 29 ----- SPI QUEUE ACQUIRED ----- #15
** 19 sendCommand READ_SINGLE_BLOCK 11956
** 10 _sendCommand 17 0x2eb4
** 11 * sending data: <Buffer 51 00 00 2e b4 ff>
** 42 TRANSFER RESULT <Buffer ff ff ff ff ff ff>
** 40 while waiting for response got <Buffer ff>
** 40 while waiting for response got <Buffer 00>
** 43 While waiting for data, got 0xff on try 100
** 32 While waiting for data, got 0xfe on try 99
** 60 ----- RELEASING SPI QUEUE ----- #15
** 668 ----- SPI QUEUE REQUESTED ----- #16
** 32 ----- SPI QUEUE ACQUIRED ----- #16
** 19 sendCommand READ_SINGLE_BLOCK 11957
** 11 _sendCommand 17 0x2eb5
** 12 * sending data: <Buffer 51 00 00 2e b5 00>
** 47 TRANSFER RESULT <Buffer ff ff ff ff ff ff>
** 46 while waiting for response got <Buffer ff>
** 43 while waiting for response got <Buffer 00>
** 45 While waiting for data, got 0xff on try 100
** 32 While waiting for data, got 0xfe on try 99
** 61 ----- RELEASING SPI QUEUE ----- #16
** 649 ----- SPI QUEUE REQUESTED ----- #17
** 31 ----- SPI QUEUE ACQUIRED ----- #17
** 19 sendCommand READ_SINGLE_BLOCK 8224
** 11 _sendCommand 17 0x2020
** 13 * sending data: <Buffer 51 00 00 20 20 00>
** 47 TRANSFER RESULT <Buffer ff ff ff ff ff ff>
** 44 while waiting for response got <Buffer ff>
** 43 while waiting for response got <Buffer 00>
** 51 While waiting for data, got 0xff on try 100
** 50 While waiting for data, got 0xfe on try 99
** 105 ----- RELEASING SPI QUEUE ----- #17
** 2051 ----- SPI QUEUE REQUESTED ----- #18
** 29 ----- SPI QUEUE ACQUIRED ----- #18
** 18 sendCommand WRITE_BLOCK 8224
** 11 _sendCommand 24 0x2020
** 11 * sending data: <Buffer 58 00 00 20 20 00>
** 43 TRANSFER RESULT <Buffer ff ff ff ff ff ff>
** 39 while waiting for response got <Buffer ff>
** 39 while waiting for response got <Buffer 00>
** 21 Sending data packet to card.
** 83 Data response was: <Buffer e5 00>
** 20 Waiting for idle, 100 more tries.
** 28 Waiting for idle, 99 more tries.
** 29 Waiting for idle, 98 more tries.
** 49 ----- RELEASING SPI QUEUE ----- #18
** 447 ----- SPI QUEUE REQUESTED ----- #19
** 28 ----- SPI QUEUE ACQUIRED ----- #19
** 19 sendCommand WRITE_BLOCK 11957
** 10 _sendCommand 24 0x2eb5
** 11 * sending data: <Buffer 58 00 00 2e b5 00>
** 43 TRANSFER RESULT <Buffer ff ff ff ff ff ff>
** 39 while waiting for response got <Buffer ff>
** 39 while waiting for response got <Buffer 00>
** 20 Sending data packet to card.
** 83 Data response was: <Buffer e5 00>
** 21 Waiting for idle, 100 more tries.
** 28 Waiting for idle, 99 more tries.
** 28 Waiting for idle, 98 more tries.
** 49 ----- RELEASING SPI QUEUE ----- #19
** 53 ----- SPI QUEUE REQUESTED ----- #20
** 28 ----- SPI QUEUE ACQUIRED ----- #20
** 19 sendCommand READ_SINGLE_BLOCK 11958
** 10 _sendCommand 17 0x2eb6
** 11 * sending data: <Buffer 51 00 00 2e b6 00>
** 43 TRANSFER RESULT <Buffer ff ff ff ff ff ff>
** 39 while waiting for response got <Buffer ff>
** 39 while waiting for response got <Buffer 00>
** 40 While waiting for data, got 0xff on try 100
** 30 While waiting for data, got 0xfe on try 99
** 56 ----- RELEASING SPI QUEUE ----- #20
** 45 ----- SPI QUEUE REQUESTED ----- #21
** 29 ----- SPI QUEUE ACQUIRED ----- #21
** 18 sendCommand WRITE_BLOCK 11958
** 11 _sendCommand 24 0x2eb6
** 12 * sending data: <Buffer 58 00 00 2e b6 9e>
** 45 TRANSFER RESULT <Buffer ff ff ff ff ff ff>
** 42 while waiting for response got <Buffer ff>
** 41 while waiting for response got <Buffer 00>
** 22 Sending data packet to card.
** 87 Data response was: <Buffer e5 00>
** 21 Waiting for idle, 100 more tries.
** 30 Waiting for idle, 99 more tries.
** 30 Waiting for idle, 98 more tries.
** 54 ----- RELEASING SPI QUEUE ----- #21
** 292 ----- SPI QUEUE REQUESTED ----- #22
** 31 ----- SPI QUEUE ACQUIRED ----- #22
** 20 sendCommand WRITE_BLOCK 18860
** 11 _sendCommand 24 0x49ac
** 12 * sending data: <Buffer 58 00 00 49 ac 00>
** 45 TRANSFER RESULT <Buffer ff ff ff ff ff ff>
** 43 while waiting for response got <Buffer ff>
** 39 while waiting for response got <Buffer 00>
** 22 Sending data packet to card.
** 92 Data response was: <Buffer e5 00>
** 22 Waiting for idle, 100 more tries.
** 32 Waiting for idle, 99 more tries.
** 31 Waiting for idle, 98 more tries.
** 55 ----- RELEASING SPI QUEUE ----- #22
** 30 ----- SPI QUEUE REQUESTED ----- #23
** 32 ----- SPI QUEUE ACQUIRED ----- #23
** 20 sendCommand READ_SINGLE_BLOCK 18861
** 10 _sendCommand 17 0x49ad
** 12 * sending data: <Buffer 51 00 00 49 ad 72>
** 44 TRANSFER RESULT <Buffer ff ff ff ff ff ff>
** 42 while waiting for response got <Buffer ff>
** 40 while waiting for response got <Buffer 00>
** 42 While waiting for data, got 0xff on try 100
** 31 While waiting for data, got 0xfe on try 99
** 62 ----- RELEASING SPI QUEUE ----- #23
** 28 ----- SPI QUEUE REQUESTED ----- #24
** 33 ----- SPI QUEUE ACQUIRED ----- #24
** 21 sendCommand WRITE_BLOCK 18861
** 11 _sendCommand 24 0x49ad
** 13 * sending data: <Buffer 58 00 00 49 ad ff>
** 45 TRANSFER RESULT <Buffer ff ff ff ff ff ff>
** 44 while waiting for response got <Buffer ff>
** 42 while waiting for response got <Buffer 00>
** 22 Sending data packet to card.
** 90 Data response was: <Buffer e5 00>
** 22 Waiting for idle, 100 more tries.
** 32 Waiting for idle, 99 more tries.
** 31 Waiting for idle, 98 more tries.
** 60 ----- RELEASING SPI QUEUE ----- #24
** 401 ----- SPI QUEUE REQUESTED ----- #25
** 71 ----- SPI QUEUE ACQUIRED ----- #25
** 42 sendCommand WRITE_BLOCK 11957
** 22 _sendCommand 24 0x2eb5
** 28 * sending data: <Buffer 58 00 00 2e b5 05>
** 93 TRANSFER RESULT <Buffer ff ff ff ff ff ff>
** 92 while waiting for response got <Buffer ff>
** 84 while waiting for response got <Buffer 00>
** 40 Sending data packet to card.
** 182 Data response was: <Buffer e5 00>
** 34 Waiting for idle, 100 more tries.
** 60 Waiting for idle, 99 more tries.
** 57 Waiting for idle, 98 more tries.
** 52 ----- RELEASING SPI QUEUE ----- #25
complete: 15.370s
```
which works fine but with 1024 bytes

```
writing ikb8ulrw.bin with 1024 bytes
** 5844 ----- SPI QUEUE REQUESTED ----- #26
** 69 ----- SPI QUEUE ACQUIRED ----- #26
** 39 sendCommand WRITE_BLOCK 8224
** 22 _sendCommand 24 0x2020
** 28 * sending data: <Buffer 58 00 00 20 20 ff>
** 90 TRANSFER RESULT <Buffer ff ff ff ff ff ff>
** 92 while waiting for response got <Buffer ff>
** 84 while waiting for response got <Buffer 00>
** 40 Sending data packet to card.
** 183 Data response was: <Buffer e5 00>
** 41 Waiting for idle, 100 more tries.
** 47 Waiting for idle, 99 more tries.
** 29 Waiting for idle, 98 more tries.
** 52 ----- RELEASING SPI QUEUE ----- #26
** 468 ----- SPI QUEUE REQUESTED ----- #27
** 29 ----- SPI QUEUE ACQUIRED ----- #27
** 19 sendCommand WRITE_BLOCK 11958
** 10 _sendCommand 24 0x2eb6
** 12 * sending data: <Buffer 58 00 00 2e b6 00>
** 42 TRANSFER RESULT <Buffer ff ff ff ff ff ff>
** 40 while waiting for response got <Buffer ff>
** 39 while waiting for response got <Buffer 00>
** 21 Sending data packet to card.
** 83 Data response was: <Buffer e5 00>
** 21 Waiting for idle, 100 more tries.
** 28 Waiting for idle, 99 more tries.
** 28 Waiting for idle, 98 more tries.
** 50 ----- RELEASING SPI QUEUE ----- #27
** 280 write blocks 18924
** 10 ----- SPI QUEUE REQUESTED ----- #28
** 28 ----- SPI QUEUE ACQUIRED ----- #28
** 19 sendCommand SET_WR_BLOCK_ERASE_COUNT 2
** 11 _sendCommand 55 0x0
** 11 * sending data: <Buffer 77 00 00 00 00 90>
** 42 TRANSFER RESULT <Buffer ff ff ff ff ff ff>
** 40 while waiting for response got <Buffer ff>
** 39 while waiting for response got <Buffer 00>
** 21 Re-syncronizing SPI bus before new command
** 28 _sendCommand 23 0x2
** 12 * sending data: <Buffer 57 00 00 00 02 00>
** 42 TRANSFER RESULT <Buffer ff ff ff ff ff ff>
** 40 while waiting for response got <Buffer ff>
** 39 while waiting for response got <Buffer 00>
** 29 sendCommand WRITE_MULTIPLE_BLOCK 18924
** 10 Re-syncronizing SPI bus before new command
** 29 _sendCommand 25 0x49ec
** 11 * sending data: <Buffer 59 00 00 49 ec 46>
** 43 TRANSFER RESULT <Buffer ff ff ff ff ff ff>
** 39 while waiting for response got <Buffer ff>
** 39 while waiting for response got <Buffer 00>
** 25 Sending data packet to card.
** 83 Data response was: <Buffer e5 00>
** 20 Waiting for idle, 100 more tries.
** 28 Waiting for idle, 99 more tries.
** 28 Waiting for idle, 98 more tries.
** 28 Waiting for idle, 97 more tries.
** 29 Waiting for idle, 96 more tries.
** 28 Waiting for idle, 95 more tries.
** 28 Waiting for idle, 94 more tries.
** 28 Waiting for idle, 93 more tries.
** 28 Waiting for idle, 92 more tries.
** 28 Waiting for idle, 91 more tries.
** 28 Waiting for idle, 90 more tries.
** 29 Waiting for idle, 89 more tries.
** 28 Waiting for idle, 88 more tries.
** 28 Waiting for idle, 87 more tries.
** 28 Waiting for idle, 86 more tries.
** 28 Waiting for idle, 85 more tries.
** 28 Waiting for idle, 84 more tries.
** 28 Waiting for idle, 83 more tries.
** 29 Waiting for idle, 82 more tries.
** 28 Waiting for idle, 81 more tries.
** 28 Waiting for idle, 80 more tries.
** 28 Waiting for idle, 79 more tries.
** 28 Waiting for idle, 78 more tries.
** 28 Waiting for idle, 77 more tries.
** 28 Waiting for idle, 76 more tries.
** 29 Waiting for idle, 75 more tries.
** 28 Waiting for idle, 74 more tries.
** 28 Waiting for idle, 73 more tries.
** 28 Waiting for idle, 72 more tries.
** 28 Waiting for idle, 71 more tries.
** 28 Waiting for idle, 70 more tries.
** 29 Waiting for idle, 69 more tries.
** 28 Waiting for idle, 68 more tries.
** 28 Waiting for idle, 67 more tries.
** 28 Waiting for idle, 66 more tries.
** 28 Waiting for idle, 65 more tries.
** 28 Waiting for idle, 64 more tries.
** 29 Waiting for idle, 63 more tries.
** 28 Waiting for idle, 62 more tries.
** 28 Waiting for idle, 61 more tries.
** 28 Waiting for idle, 60 more tries.
** 28 Waiting for idle, 59 more tries.
** 29 Waiting for idle, 58 more tries.
** 28 Waiting for idle, 57 more tries.
** 28 Waiting for idle, 56 more tries.
** 28 Waiting for idle, 55 more tries.
** 28 Waiting for idle, 54 more tries.
** 28 Waiting for idle, 53 more tries.
** 29 Waiting for idle, 52 more tries.
** 28 Waiting for idle, 51 more tries.
** 28 Waiting for idle, 50 more tries.
** 28 Waiting for idle, 49 more tries.
** 28 Waiting for idle, 48 more tries.
** 29 Waiting for idle, 47 more tries.
** 28 Waiting for idle, 46 more tries.
** 28 Waiting for idle, 45 more tries.
** 28 Waiting for idle, 44 more tries.
** 28 Waiting for idle, 43 more tries.
** 28 Waiting for idle, 42 more tries.
** 29 Waiting for idle, 41 more tries.
** 28 Waiting for idle, 40 more tries.
** 30 Waiting for idle, 39 more tries.
** 31 Waiting for idle, 38 more tries.
** 30 Waiting for idle, 37 more tries.
** 29 Waiting for idle, 36 more tries.
** 30 Waiting for idle, 35 more tries.
** 30 Waiting for idle, 34 more tries.
** 30 Waiting for idle, 33 more tries.
** 29 Waiting for idle, 32 more tries.
** 29 Waiting for idle, 31 more tries.
** 30 Waiting for idle, 30 more tries.
** 30 Waiting for idle, 29 more tries.
** 31 Waiting for idle, 28 more tries.
** 30 Waiting for idle, 27 more tries.
** 30 Waiting for idle, 26 more tries.
** 30 Waiting for idle, 25 more tries.
** 30 Waiting for idle, 24 more tries.
** 30 Waiting for idle, 23 more tries.
** 30 Waiting for idle, 22 more tries.
** 31 Waiting for idle, 21 more tries.
** 30 Waiting for idle, 20 more tries.
** 30 Waiting for idle, 19 more tries.
** 31 Waiting for idle, 18 more tries.
** 29 Waiting for idle, 17 more tries.
** 29 Waiting for idle, 16 more tries.
** 31 Waiting for idle, 15 more tries.
** 31 Waiting for idle, 14 more tries.
** 31 Waiting for idle, 13 more tries.
** 32 Waiting for idle, 12 more tries.
** 31 Waiting for idle, 11 more tries.
** 31 Waiting for idle, 10 more tries.
** 32 Waiting for idle, 9 more tries.
** 30 Waiting for idle, 8 more tries.
** 31 Waiting for idle, 7 more tries.
** 29 Waiting for idle, 6 more tries.
** 31 Waiting for idle, 5 more tries.
** 29 Waiting for idle, 4 more tries.
** 29 Waiting for idle, 3 more tries.
** 30 Waiting for idle, 2 more tries.
** 30 Waiting for idle, 1 more tries.
** 29 Waiting for idle, 0 more tries.
** 54 ----- RELEASING SPI QUEUE ----- #28
** 237 ----- SPI QUEUE REQUESTED ----- #29
** 30 ----- SPI QUEUE ACQUIRED ----- #29
** 21 sendCommand WRITE_BLOCK 11958
** 11 _sendCommand 24 0x2eb6
** 13 * sending data: <Buffer 58 00 00 2e b6 9f>
** 45 TRANSFER RESULT <Buffer c0 00 00 00 00 00>
** 42 while waiting for response got <Buffer 00>
** 22 Sending data packet to card.
** 94 Data response was: <Buffer ff ff>
** 70 ----- RELEASING SPI QUEUE ----- #29
complete: 12.040s Error: No more tries left waiting for idle.
```
it fails.

also: between running my test code with `tessel run` i have to release and reinject the sd card, or i'll get the error `Error: Unknown or missing card. Error: Timed out waiting for reponse.`.

**Comment by yetzt** [comment]
so far i found out, with 1023 bytes, `fatfs` calls `writeSectors` with blocks of 512 bytes only, which calls `writeBlock` and works fine:

```
writing ikb97aff.bin with 1023 bytes
writing 512 bytes to sector 32
writing 512 bytes to sector 3766
writing 512 bytes to sector 10860
writing 512 bytes to sector 10861
writing 512 bytes to sector 3766
complete: 12.006s
```
on the other hand, with 1024 bytes `writeSectors` is called with a block of 1024 bytes, which calls `writeBlocks`, which uses the command `WRITE_MULTIPLE_BLOCK` and fails.

**Comment by yetzt** [comment]
i tried it with another sd card, no problem there. i'll keep investigating.

**Comment by yetzt** [comment]
sadly, my card behaviour seems, as far as i understand it, to break the spec. there does not seem to be an indicator when such behaviour is to be expected of an sdcard. even fetching the CSD and looking at WRITE_BL_LEN is inconclusive. therefore i made this humble workaround, which you may merge.

**Comment by natevw** [comment]
Yeah, a weird card was going to be my guess. Hope to look at your patch early next week, thanks!

**Comment by yetzt** [comment]
thanks, @natevw 

i did some testing regarding to speed. it's generally dismal, but WRITE_BLOCK is considerably slower.

```
writing 32 bytes with WRITE_MULTI_BLOCK       4.889s
writing 32 bytes with WRITE_BLOCK             7.164s

writing 64 bytes with WRITE_MULTI_BLOCK       4.095s
writing 64 bytes with WRITE_BLOCK             5.910s

writing 128 bytes with WRITE_MULTI_BLOCK      3.543s
writing 128 bytes with WRITE_BLOCK            6.024s

writing 256 bytes with WRITE_MULTI_BLOCK      4.401s
writing 256 bytes with WRITE_BLOCK            6.117s

writing 512 bytes with WRITE_MULTI_BLOCK      3.603s
writing 512 bytes with WRITE_BLOCK            6.050s

writing 1024 bytes with WRITE_MULTI_BLOCK     4.873s
writing 1024 bytes with WRITE_BLOCK           6.491s

writing 2048 bytes with WRITE_MULTI_BLOCK     6.370s
writing 2048 bytes with WRITE_BLOCK           7.236s

writing 4096 bytes with WRITE_MULTI_BLOCK     6.604s
writing 4096 bytes with WRITE_BLOCK           9.996s

writing 8192 bytes with WRITE_MULTI_BLOCK     8.669s
writing 8192 bytes with WRITE_BLOCK          11.594s

writing 16384 bytes with WRITE_MULTI_BLOCK   13.545s
writing 16384 bytes with WRITE_BLOCK         18.576s

writing 32768 bytes with WRITE_MULTI_BLOCK   21.185s
writing 32768 bytes with WRITE_BLOCK         30.482s

writing 65536 bytes with WRITE_MULTI_BLOCK   38.651s
writing 65536 bytes with WRITE_BLOCK         55.578s

writing 131072 bytes with WRITE_MULTI_BLOCK  72.096s
writing 131072 bytes with WRITE_BLOCK       105.247s

writing 262144 bytes with WRITE_MULTI_BLOCK 143.730s
writing 262144 bytes with WRITE_BLOCK       205.424s
```

**Comment by natevw** [comment]
Now published as sdcard@0.8.9. Very belated thanks for your detective work and clean patch!
